ecs-cli/modules/config/ami/ami.go

```diff
@@ -45,11 +59,55 @@ func NewStaticAmiIds() ECSAmiIds {
 	return &staticAmiIds{regionToId: regionToId}
 }
 
+// Get returns the AMI ID for a given region
+// It returns the AMI ID that it finds in:
+// 	1. The S3 bucket
+// 	2. The cn-north-1 mirror of the main S3 bucket
+// 	3. The hardcoded list (which may be slightly out of date. )
+// (#2 helps users in china who may be blocked from the main bucket)
 func (c *staticAmiIds) Get(region string) (string, error) {
+	// first try to pull from the main bucket
+	if id, err := getIDFromS3(mainBucketS3URL+region, maxRetries); err == nil {
+		return id, nil
+	}
+
+	// if we could not reach the main bucket, try the bucket in China
+	if id, err := getIDFromS3(mainBucketS3URL+region, maxRetries); err == nil {
```

Comment:
bug: `mainBucketS3URL` should be `cnNorth1BucketS3URL`

I would suggest to figure out what region it is and base on the region call different endpoint.
```
var url string
switch region
case cn-north-1:
   url = cnNorth1BucketS3URL + region
default:
   url = mainBucketS3URL + region

// getIDFromS3
```